### DISCUSSION TITLE
Inconsistent button text colors

### DISCUSSION THREAD

**Discussion by sohaha** [saadeghi/daisyui#2046] [Q&A] [answered]
After looking at the code, the color of the button seems to be dynamically generated by generateForegroundColorFrom.

<img width="942" alt="image" src="https://github.com/saadeghi/daisyui/assets/18098083/839ba115-e8f2-4752-a19a-448a89e09451">

The text is black and white. If you want to unify many colors but cannot use them, can you expose the percentage and configure it yourself?

**Comment by saadeghi** [comment, marked as answer]
It's not inconsistent. It's not supposed to be black or white. Each theme has their own color for background and foreground of buttons. They are just design decisions for built-in themes.  
You can also [customize the built-in themes](https://daisyui.com/docs/themes/#-7) or [create your own theme](https://daisyui.com/theme-generator/) to have full control of all colors.

Here you can find all the color names: https://daisyui.com/docs/colors/

`generateForegroundColorFrom` is an internal function, for when a theme doesn't have the optional colors like `primary-content`, `secondary-content`, and [other `*-content` color names](https://daisyui.com/docs/colors/)  
If those optional colors are not defined [in the theme](https://daisyui.com/docs/themes/#-4), that function generates a color with good contrast automatically. It uses `colord` npm package and mixes the background color with black or white with `0.8` ratio. But you don't need to deal with that function or that package. You can simply add [all the color names](https://daisyui.com/docs/colors/) to your theme and give them values.

**Reply by sohaha** [reply]
```js
module.exports = {
  //...
  daisyui: {
    themes: [
      {
        mytheme: {
          '--pc': '0 0% 100%',
          '--nc': '0 0% 100%',
          '--wac': '0 0% 100%',
          '--sc': '0 0% 100%',
        },
      },
    ],
  },
}
```
The above configuration, only sc takes effect, others are invalid

**Reply by sohaha** [reply]
I want to unify the color of the button text, is there a way to achieve it?

**Comment by saadeghi** [comment]
> The above configuration, only sc takes effect, others are invalid

Use color names, not the CSS variable. Read the docs: https://daisyui.com/docs/themes/#-4

> I want to unify the color of the button text, is there a way to achieve it?

Yes, you just need to specify all the `content-*` colors like this:  
https://play.tailwindcss.com/y8qzipd90Z?file=config
```js
  daisyui: {
    themes: [
      {
        mytheme: {
          primary: '#e56a57',
          secondary: '#c2ef8b',
          accent: '#facc15',
          neutral: '#222028',
          "primary-content": '#000000',    // 👈
          "secondary-content": '#000000',  // 👈
          "accent-content": '#000000',     // 👈
          "neutral-content": '#000000',    // 👈
          'base-100': '#ffffff',
          info: '#4f46e5',
          success: '#57e0a5',
          warning: '#d1aa10',
          error: '#e76e89',
        },
      },
    ],
  },
```

**Reply by sohaha** [reply]
thank you for your help
